src/lambda/models/samLambdaRuntime.ts

```diff
@@ -92,7 +112,9 @@ export function getDependencyManager(runtime: Runtime): DependencyManager[] {
 }
 
 export function getFamily(runtime: string): RuntimeFamily {
-    if (nodeJsRuntimes.has(runtime)) {
+    if (deprecatedRuntimes.has(runtime)) {
+        handleDeprecatedRuntime(runtime)
```

Comment:
good idea. 

alternative to throwing an error, is to return `RuntimeFamily.deprecated` here, then the caller (either "Create new SAM app" command or the "Run/Debug" entrypoint) would handle it explicitly. That could avoid potential crashes, since the contract of these getters was previously non-throwing.

Reply:
What do you think about `return undefined`? Error is harsh but returning a runtime family doesn't necessarily mean we'll remember to handle it (undefined is a clearer signal). I do like emitting the message from a single function though, I can't imagine we'll be changing that messaging all that much between applications.

Reply:
If you make it `undefined` you'll have to add code to check for undefined anyway. Plus callers now have to figure out that undefined is equivalent to a deprecated runtime, whereas returning an enum is self-documenting.

Reply:
The enum isn't really self-documenting if you just pipe it into a function looking for a RuntimeFamily without thinking (which is also dangerous because we're mainly handling those through switches, so it'll just fall into the default case). `undefined` forces action through compiler hints.

Reply:
It's perfectly valid to switch on `undefined` though. You have to manually check the return type to see that it can potentially return undefined, then handle it separately. This only works when upstream functions are calling `getFamily` then pass the results somewhere else. But at that point just union the return type with 'deprecated', you get the same effect.

Reply:
not a blocker. can throw an error for now